torch/distributions/kl.py

```diff
@@ -310,7 +313,11 @@ def _kl_poisson_poisson(p, q):
 def _kl_transformed_transformed(p, q):
     if p.transforms != q.transforms:
         raise NotImplementedError
-    return kl_divergence(p.base_dist, q.base_dist)
+    if p.event_shape != q.event_shape:
+        raise NotImplementedError
+    event_dim = len(p.event_shape)
```

Comment:
I think you'll need to account for cases where `p.base_dist` is itself multivariate:
```py
extra_event_dim = len(p.event_shape) - len(p.base_dist.event_shape)
...
_sum_rightmost(base_kl_divergence, extra_event_dim)
```